oak_functions/lookup/README.md

```diff
@@ -0,0 +1,42 @@
+# Lookup
+
+We want to maintain the following invariants on the key/value lookups in the
+lookup data loaded into Oak Functions.
+
+## Invariant: Consistent view on lookup data.
+
+When the lookup data is updated in the background, key/value lookups of requests
+which arrived after the update will return values from the new lookup data, but
+key/value lookups of requests which arrived before the update will return values
+from the old lookup data.
+
+In particular, within a request for two key/value lookups of the same key Oak
+Functions will return the same value.
+
+_Reasoning_: We want a consistent view of the lookup data within the life time
+of a request. In the worst case, this can lead to _n_ copies of lookup data for
+_n_ requests running in parallel, but we expect short-lived requests and
+low-frequency updates of lookup data.
+
+## Invariant: Fully loaded lookup data.
+
+No key/value lookups of requests are served until the lookup data is completely
```

Comment:
Perhaps highlight that this is the "initial lookup data". Also no requests should be served at all. This seems to imply that the requests are handled but they cannot do lookup data.

Reply:
True: Why do we not want to serve requests which don't lookup lookup data?

Reply:
@conradgrobler I added a description and updated the explanation on why I think we cannot serve any request. Can you check?

Reply:
I don't see the changes. Have you pushed them to github yet?

We don't analyse the request-handling Wasm, so we don't know whether it will try to access lookup data or not. In general though, if the lookup data extension is enabled on a server we assume the Wasm code will do data lookups as part of every request. If we start serving requests and become included in the set of instance used by the load-balancer we will start getting traffic. If we detect that the Wasm logic is trying to lookup up data we have to generate an error, which would mean the request is invalid and the client would have to retry. I think it is better for clients overall if their requests take slightly longer because of one fewer server serving requests rather than some being sent to a server that cannot possibly serve a valid response, forcing them to retry.

Reply:
Sorry, forgot to push, thank you!